### PR TITLE
CIP-0108? | Governance Metadata - Governance Actions

### PR DESCRIPTION
**Pull request by Ryun1**
The conway ledger era ushers in on-chain governance for Cardano via [CIP-1694 | A First Step Towards On-Chain Decentralized Governance](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md), with the addition of many new on-chain governance artifacts.
Some of these artifacts support the linking off-chain metadata, as a way to provide context.

The [CIP-100 | Governance Metadata](https://github.com/cardano-foundation/CIPs/pull/556) standard provides a base framework for how all off-chain governance metadata should be formed and handled.
But this is intentionally limited in scope, so we expand it here defining fields for governance action metadata.

#### TODO
- [x] Run three workshops via Intersect Governance Metadata Working Group
- [x] Flesh out rationale section
---

[📜 Rendered on fork](https://github.com/Ryun1/CIPs/blob/governance-metadata-actions/CIP-0108/README.md)

### CODE REVIEW

**Comment by Ryun1** [comment]
For those interested; 

I have put this proposal on the Triage agenda for CIP Editors Call # 78 (December 12th 4pm UTC), held in [The CIP Editors Meetings Discord](https://discord.gg/jvgrKbER), see [agenda](https://hackmd.io/@cip-editors/78) and [discord event](https://discord.com/events/971785110770831360/1179107190796210316).

File: CIP-governance-metadata-actions/README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+CIP: ?
```
Comment (rphair) [inline CIP-governance-metadata-actions/README.md:2]:
```suggestion
CIP: 108
```

**Review by rphair — COMMENTED** [review]
Assigning CIP number based on unanimously positive response at yesterday's CIP meeting.

**Comment by Ryun1** [comment]
This proposal is looking quite good at the moment.
There is only some minor tidy up left as well as writing some examples.
I will finish do the final tidy once https://github.com/cardano-foundation/CIPs/pull/748 is settled.

File: CIP-0108/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+### For all participants
+By standardizing off-chain metadata formats we facilitate interoperability for tooling which creates and/or renders metadata attached to governance actions.
+This intern promotes a rich user experience between tooling.
+This is good for all governance participants.
+
+## Specification
+Although there are seven types of governance action defined via CIP-1694, we focus this proposal on defining core properties which must be attached to all types.
+We leave room for future standards to refine and specialize further to cater more specific for each type of governance action.
+
+### Markdown Text Styling
```
[the remark this answers is not in the document]
Comment (Crypto2099) [reply CIP-0108/README.md:88]: Supporting markdown means that we can allow rich text formatting without much concern that there will be malicious code injected (explorers can explicitly strip all HTML tags for safety or allow only a subset of tags). Generally speaking two blank lines equals a carriage return in Markdown but we could also declare a list of minimum HTML tags that can be supported such as the `<br />` tag.

[the remark this answers is not in the document]
Comment (Ryun1) [reply CIP-0108/README.md:88]: Lets do it!

[the remark this answers is not in the document]
Comment (rphair) [reply CIP-0108/README.md:88]: @Ryun1 @Crypto2099 yes & by **not** including minimal HTML tags like `<i>` and `<b>` on the list we can promote standardisation of document format like Markdown's corresponding`*` (or `_`) and `**`.

File: CIP-0108/README.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+- i.e "by decreasing X parameter by Y we increase Ada earned by SPOs, thus incentivising more people to become SPOs, leading to a more diverse network"
+- This SHOULD provide evidence of consensus within the community and discuss significant objections or concerns raised during the discussion.
+- This SHOULD include discussion of alternative solutions and related topics/ governance actions.
+- This SHOULD include any recommendations made by relevant organizations or committees.
+
+#### `references`
+- We extend CIP-100's references field.
+- This SHOULD NOT support markdown text styling.
+- To be an OPTIONAL set of objects, using the `@set` property.
+- Each object MUST have a `title` field to describe the reference, such as; "blog - Why we must continue to fund Catalyst".
```
Comment (Ryun1) [inline CIP-0108/README.md:140]:
```suggestion
- Each object MUST have a `label` field to describe the reference, such as; "blog - Why we must continue to fund Catalyst".
```

small fix

Comment (Ryun1) [reply CIP-0108/README.md]: Good spot @jdyczka

File: CIP-0108/README.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+---
+
+## Abstract
+The Conway ledger era ushers in on-chain governance for Cardano via [CIP-1694 | A First Step Towards On-Chain Decentralized Governance](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md), with the addition of many new on-chain governance artifacts.
+Some of these artifacts support the linking off-chain metadata, as a way to provide context.
+
+The [CIP-100 | Governance Metadata](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0100) standard provides a base framework for how all off-chain governance metadata should be formed and handled.
+But this is intentionally limited in scope, so that it can be expanded upon by more specific subsequent CIPs.
+
+This proposal aims to provide a specification for off-chain metadata vocabulary can be used to give context to governance actions, CIP-100.
```
Comment (Ryun1) [inline CIP-0108/README.md:24]:
```suggestion
This proposal aims to provide a specification for off-chain metadata vocabulary that can be used to give context to governance actions.
```

grammar fix

File: CIP-0108/cip-0108.common.schema.json [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            "description": "An author endorsing the content of a metadata document (CIP-100)",
+            "type": "object",
+            "required": ["name", "witness"],
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "title": "Name"
+                },
+                "witness": {
+                    "$ref": "#/definitions/witness"
```
Comment (Crypto2099) [inline CIP-0108/cip-0108.common.schema.json:43]: You do not define witness in this document? I believe that instead this needs to be a link to CIP-100 witness definition? By the same token, the entire author/authors blocks could probably be shunted off to the CIP-100 definition unless there is something being customized here in CIP-108?

Comment (Ryun1) [reply CIP-0108/cip-0108.common.schema.json:43]: good catch
I wasnt sure if I should add the CIP-100 related definitions to the common schema file or not

Comment (Crypto2099) [reply CIP-0108/cip-0108.common.schema.json:43]: Generally speaking they are fine to include as long as you reference them using a ref link that points to the canonical source. This just tells the schema checker to look at that link for that piece of data

**Comment by Ryun1** [comment]
I have significantly refactored the provided test vector's file and examples.

This should make recreating examples much easier.

**Review by rphair — APPROVED** [review]
I just went over https://github.com/cardano-foundation/CIPs/pull/632/commits/fc2fe1de6431ce4db341019e14b029151d4be466 which to me look like "finishing touches" so this seems as ready as I can declare it to be (would happily merge in the meeting in 1 hour's time if no dispute).

File: CIP-0108/README.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+- <s>How much vocabulary can be extended to other onchain governance events?</s>
+  - It is hard to predict how the scope of future standards before they have been developed.
+- <s>How to integrate custom set of HTML tags? to allow formatting of longer text fields.</s>
+  - Since CIP-100 does not intend to support rich text fields such an inclusion would not fit, so we have included such format here.
+
+## Path to Active
+
+### Acceptance Criteria
+- [ ] This standard is supported by two different tooling providers used to submit governance actions to chain.
+- [ ] This standard is supported by two different chain indexing tools, used to read and render metadata.
```
Comment (Ryun1) [inline CIP-0108/README.md:179]:
```suggestion
- [ ] This standard is supported by two different chain indexing tools, used to read and render metadata.
  - DB-Sync via [db-sync-sancho-4.1.0](https://github.com/IntersectMBO/cardano-db-sync/releases/tag/sancho-4.1.0)
```

**Review by Crypto2099 — APPROVED** [review]
Looks like this is a state where it is ready to be published. Revisions (if needed) can be amended to this document.

**Pull request merged**